matryoshka_tester/data.py

```diff
@@ -1,27 +1,108 @@
-containers = dict()
-containers["go"] = {
-    "1.16": "registry.opensuse.org/home/fcrozat/matryoshka/containerfile/bci/golang:1.16",
-    "1.15": "registry.opensuse.org/home/fcrozat/matryoshka/containerfile/bci/golang:1.15",
-}
-containers["node"] = {
-    "15": "registry.opensuse.org/home/fcrozat/matryoshka/containers_node15/node15:latest",
-    "14": "registry.opensuse.org/home/fcrozat/matryoshka/containers_node14/node14:latest",
-    "12": "registry.opensuse.org/home/fcrozat/matryoshka/containers_node12/node12:latest",
-}
-containers["openjdk-devel"] = {
-    "16": "registry.opensuse.org/devel/bci/images/bci/openjdk-devel:16",
-    "15": "registry.opensuse.org/devel/bci/images/bci/openjdk-devel:15",
-    "14": "registry.opensuse.org/devel/bci/images/bci/openjdk-devel:14",
-    "11": "registry.opensuse.org/devel/bci/images/bci/openjdk-devel:11",
-}
-containers["openjdk"] = {
-    "16": "registry.opensuse.org/devel/bci/images/bci/openjdk:16",
-    "15": "registry.opensuse.org/devel/bci/images/bci/openjdk:15",
-    "14": "registry.opensuse.org/devel/bci/images/bci/openjdk:14",
-    "11": "registry.opensuse.org/devel/bci/images/bci/openjdk:11",
-}
-containers["python"] = {
-    "3.9": "registry.opensuse.org/home/fcrozat/matryoshka/containers_python39/python39:latest",
-    "3.8": "registry.opensuse.org/home/fcrozat/matryoshka/containers_python38/python38:latest",
-    "3.6": "registry.opensuse.org/home/fcrozat/matryoshka/containers_python36/python36:latest",
+from dataclasses import dataclass
```

Comment:
What I liked in this file, was the very simple way for a non-experienced user to iterate here, add its own container, change it. I am not sure if this is still as readable (it is okay for me, but I admit it's not as clear).

If I were to rebuild this, I would probably make sure that we have data in a certain form, and (un)marshall it into the proper python objects (this would be the way to build the `containers` dict [maybe we could move this to a list?]). The objects could be `Container`s, for which we could have different implementations, based on the input data. e.g. we could inherit the container for specific language containers (`GoContainer`?). I feel like this would make more sense and be more readable.  WDYT? 

In a certain way, I wanted to separate the _data_ from the code.

Side note: ContainerConfig would be removed too if we had this, and make the whole purpose of the ContainerConfig (type validation) far more clear, IMO.

Reply:
Yeah, I must confess that the part of my code that I really hate is how it makes the registry URL addition super confusing. Unmarshalling it sounds way better, so I'll give that a try.